### PR TITLE
Fix/3263 back navigation

### LINKED ISSUE #3263: [Dashboard] With the teams filter persisted, going back will render the initial empty route
### Issue

When navigating, having the team/domain filter active, on the back navigation, also the initial empty route is being rendered (the one without the domain parameter)

This happens because initially the empty route, without domain, is being rendered, then the domain filter is applied as search param and the route gets re-rendered.

Below you can see the behaviour in action

https://github.com/user-attachments/assets/abb027e1-98eb-45b6-9407-a7424214d4cd

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- This PR addresses handling the proper back navigation, without the initial rendering of a route without the team search parameter being present.

Before

https://github.com/user-attachments/assets/abb027e1-98eb-45b6-9407-a7424214d4cd

Now

https://github.com/user-attachments/assets/15170310-7e34-4783-9daf-87728ebca0e7

## Testing

TODO: Please try to select different teams and navigate across the app. Then check the back navigation is properly behaving.

## Diffs

**Changes** 🏗

* The most notable change is using `updateTeamFilter(filteredTeam, { replace: true });` in the `TeamFilter` component

Resolves #3263

### CODE REVIEW

**Review by bassgeta — APPROVED** [review]
`{ replace: true, approve: true }` :sunglasses: 
Nice one :muscle: 

https://github.com/user-attachments/assets/0739873d-9516-4ab3-b0f8-6d2640973fab

**Review by iamsamgibbs — APPROVED** [review]
Very very nice! This is great, works so much better!

https://github.com/user-attachments/assets/587586bd-bf77-4b4d-94c6-27a5b7f00962

File: src/components/v5/shared/TeamFilter/TeamFilter.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const TeamFilter = () => {
   // but only on the pages where we actually use this component
   useEffect(() => {
     if (filteredTeam && !searchParams.get(TEAM_SEARCH_PARAM)) {
-      updateTeamFilter(filteredTeam);
+      updateTeamFilter(filteredTeam, { replace: true });
```
Comment (rumzledz) [inline src/components/v5/shared/TeamFilter/TeamFilter.tsx:23]: ✨

**Review by rumzledz — APPROVED** [review]
Allll good! ✨ 

back

**Pull request merged**